### PR TITLE
Fix the try build when pushing to branches

### PR DESCRIPTION
**Pull request by mrobinson**
Make it so that all try builds go through try.yml and pass
workflow_call arguments as expected to subsequent workflows.

Co-authored-by: Samson <16504129+sagudev@users.noreply.github.com>

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they fix a CI issue.

### CODE REVIEW

**Comment by sagudev** [comment]
I will do some testing on my fork but generally looks good.

File: .github/workflows/windows.yml
Comment (sagudev) [inline .github/workflows/windows.yml:1]: This is still run on push to try-windows branch.

Comment (mrobinson) [reply .github/workflows/windows.yml:1]: Oh! Nice catch. I've removed this as well as one more stray reference to `ref_name` in `linux-wpt.yml`.

**Comment by sagudev** [comment]
try-wpt-mac does not work (probably because try labels also do not support them), but given this was never really used I think we can wait for #30694 to bring it back.

**Review by sagudev — APPROVED** [review]

**Comment by mrobinson** [comment]
> try-wpt-mac does not work (probably because try labels also do not support them), but given this was never really used I think we can wait for #30694 to bring it back.

That's true and a bit unfortunate. I will try to do a followup that uses more of your work in #30694 to allow configuring which platforms run tests.

**Comment by mrobinson** [comment]
Thanks very much for the review.

**Comment by sagudev** [comment]
> > try-wpt-mac does not work (probably because try labels also do not support them), but given this was never really used I think we can wait for #30694 to bring it back.
> 
> That's true and a bit unfortunate. I will try to do a followup that uses more of your work in #30694 to allow configuring which platforms run tests.

I plan to split some stuff from #30694 to separate PR after this one land. Mostly wpt -> wpt-sync boolean & wpt to be used from custom wpt test string (to test only specific tests) and flake8 config.

**Pull request merged**